src/concurrency/thread.rs

```diff
@@ -184,6 +188,12 @@ impl<'mir, 'tcx> Thread<'mir, 'tcx> {
 #[derive(Debug)]
 pub enum Time {
     Monotonic(Instant),
+    Virtual {
+        /// The instant for this moment.
+        instant: u64,
+        /// A reference to the time anchor used to read the current global virtual time.
+        time_anchor: Arc<AtomicU64>,
```

Comment:
Why does this need an anchor when the others do not?

I was honestly thinking this use `Monotonic`. Only the code that actually tests whether an `Instant` has passed, and that sleeps until an `Instant` passes, should need adjusting.

Reply:
Is just because you need to access the time_anchor to compute how much time is left when calling `get_wait_time`. 

Maybe I misunderstood but were you thinking about manually increasing `Instant` by `10 ns` on every tick and use that as the anchor?